### ISSUE TITLE
Document code flow for WordPress migration

### ISSUE THREAD

**Issue by vishnugopal** [Automattic/wp-calypso#65476]

**Comment by ouikhuan** [comment]
### Objective

We want to work on the improvements for our migration flow, but we'll need to have a good understanding of the codebase/codeflow before we start tackling it. 

### Approach

Here are a few key points we want to focus on:

- [x] The way we detect the site address and see if's a self-hosted WP site or a WP.com or just has a domain plan with us.
- [x] How we detect and know if a site needs a Jetpack connection and how we verify the Jetpack site
- [x] How Jetpack installation works under the hood
- [x] How the Jetpack plan works and the necessity of activate at least one plan before migration
- [x] How we verify the status of the importing process

**Comment by ouikhuan** [comment]
We've split it into three posts:
- [Site address and Jetpack detection](pcmemI-NT-p2)
- [The JPC flow](pcmemI-19Y-p2)
- [Jetpack Plans page and Import Everything](pcmemI-1a2-p2)
